xgcm/grid.py

```diff
@@ -1306,14 +1306,20 @@ def set_metrics(self, key, value, overwrite=False):
     def _get_dims_from_axis(self, da, axis):
         dim = []
         for ax in axis:
-            all_dim = self.axes[ax].coords.values()
-            matching_dim = [di for di in all_dim if di in da.dims]
-            if len(matching_dim) == 1:
-                dim.append(matching_dim[0])
+            set_ax = frozenset(_maybe_promote_str_to_list(ax))
```

Comment:
I think the promotion should be outside the loop since you're assuming that `axis` is `("X",)` and not `"X"` for example

so 
``` python
axis = _maybe_promote_str_to_list(ax)
for ax in axis:
    if ax in self.axes:
        ...
```

you don't always have to use set operations to check for membership.

Reply:
Wait I don't see what the improvement is over the old version.

Reply:
In the old version, if `ax` wasn't present in `self.axes`:
```python
for ax in axis:
    all_dim = self.axes[ax].coords.values()
```
This line would just produce an error: `KeyError: []`
I included an if statement:
```python
set_ax = frozenset(_maybe_promote_str_to_list(ax))
if set_ax.issubset(set(self.axes.keys())):
```
so that the error would have more explanation: `KeyError("Did not find axis %s from data array %s" % (ax, da.name))`

Reply:
ah nice!